Ghidra/Processors/RISCV/data/languages/riscv.instr.sinc

```diff
@@ -46,9 +46,23 @@
 @include "riscv.priv.sinc"
 @include "riscv.rvc.sinc"
 @include "riscv.rvv.sinc"
+@include "riscv.zbkx.sinc"
+@include "riscv.zknd.sinc"
+@include "riscv.zkne.sinc"
+@include "riscv.zknh.sinc"
+@include "riscv.zksed.sinc"
+@include "riscv.zksh.sinc"
+@include "riscv.zvbb.sinc"
+@include "riscv.zvkng.sinc"
+@include "riscv.zvksg.sinc"
 @include "riscv.zi.sinc"
 
-@include "riscv.custom.sinc"
+# discard generic custom instructions in favor of documented custom instructions
+#@include "riscv.custom.sinc"
+
+# Thead semi's extensions currently recognized by binutils objdump
+#   and documented in https://github.com/T-head-Semi/thead-extension-spec/releases/download/2.0.0/xthead-2022-09-05-2.0.0.pdf
+@include "riscv.xthead.sinc"
```

Comment:
This should be guarded by `ifdef`, with the `define` in a new xhead slaspec file

Reply:
That's reasonable.  Does the new xthead slaspec file get named in `riscv.ldefs` so the user can invoke it, or do you suggest we generalize `riscv.opinion` to look for the `Tag_RISCV_arch` ELF attribute, recognize the current composition of extensions,  and set finer-grain inclusion tags? 
```
Tag_RISCV_arch: "rv64i2p1_m2p0_a2p1_f2p2_d2p2_v1p0_zicsr2p0_zifencei2p0_zmmul1p0_zve32f1p0_zve32x1p0_zve64d1p0_zve64f1p0_zve64x1p0_zvl128b1p0_zvl32b1p0_zvl64b1p0_xtheadfmemidx1p0"
```

Apparently binutils 2.41 concatenates all of the `march` extensions passed to gcc, so the tail end of this attribute reads something like:

"This binary requires an ISA supporting:"

* `x` - a vendor specific Instruction Set Architecture module not currently part of a proposed standard profile
* `thead` - the lower case vendor name publishing the extension set
* `fmemidx` - the extension set module name
* `1p0` - the version of this extension set, likely `1.0`

Composing standard and vendor extension profiles raises lots of Ghidra import design questions.  Apparently questions the binutils team has already addressed - do their answers work for Ghidra?